matkt/pantheon
# [PAN-2967] EthGetProof Review Touch Ups

## PR description

This PR includes a few cosmetic touch-ups:
* Use `UInt256` for raw (unhashed) storage keys
* Simplify `WorldStateProof` by removing generics 

As well as some bug / performance fixes:
* Always include the root node in the storage proof
* Only look up account once when producing proof

And some test modifications:
*  `WorldStateProofProviderTest`
   * Don't use mocks instead use real storage instances
   * Expand test case to include proof with with non-empty storage keys
* `AbstractMerklePatriciaTrieTest`
   * Create abstract base class to avoid duplicate tests for various trie classes
   * Add / modify tests to check that root node is included in proofs
* `EthGetProofTest`
   * Add test case where world state is unavailable
